images/macos/provision/core/openjdk.sh

```diff
@@ -1,28 +1,48 @@
 #!/bin/bash -e -o pipefail
 source ~/utils/utils.sh
 
+JAVA_TOOLCACHE_PATH=$AGENT_TOOLSDIRECTORY/Java_Adoptium_jdk
+
 createEnvironmentVariable() {
-    local VARIABLE_NAME=$1
-    local JAVA_VERSION=$2
-    if [[ $JAVA_VERSION == "8" ]]; then
-        JAVA_VERSION="1.${JAVA_VERSION}"
+    local JAVA_VERSION=$1
+    local JAVA_PATH=$2
+
+    local JAVA_HOME_PATH=$JAVA_PATH/Contents/Home
+    if [[ $JAVA_VERSION == $JAVA_DEFAULT ]]; then
+        echo "export JAVA_HOME=${JAVA_HOME_PATH}" >> "${HOME}/.bashrc"
+        export PATH="$JAVA_HOME_PATH:$PATH"
     fi
-    local JAVA_PATH=$(/usr/libexec/java_home -v${JAVA_VERSION})
-    echo "export ${VARIABLE_NAME}=${JAVA_PATH}" >> "${HOME}/.bashrc"
+    echo "export JAVA_HOME_${JAVA_VERSION}_X64=${JAVA_HOME_PATH}" >> "${HOME}/.bashrc"
 }
 
-brew tap AdoptOpenJDK/openjdk
+installJavaFromAdoptOpenJDK() {
+    local JAVA_VERSION=$1
+
+    javaRelease=$(curl -s "https://api.adoptopenjdk.net/v3/assets/latest/${JAVA_VERSION}/hotspot" | jq -r '[.[] | select(.binary.os=="mac")][0]')
```

Comment:
I think we should extend `select(.binary.os=="mac")` condition. This JSON file contains at least different java-package versions (`jdk` and `jre`).
Ideally, we need to filter by os, architecture, image_type.

Also suggest to split curl and jq invocations to a few lines.